model/build-processor/src/main/java/org/keycloak/models/map/processor/Util.java

```diff
@@ -46,7 +47,7 @@
 public class Util {
 
     private static final Set<String> LIST_TYPES = Set.of(List.class.getCanonicalName(), ArrayList.class.getCanonicalName(), LinkedList.class.getCanonicalName());
-    private static final Set<String> SET_TYPES = Set.of(Set.class.getCanonicalName(), TreeSet.class.getCanonicalName(), HashSet.class.getCanonicalName(), LinkedHashSet.class.getCanonicalName());
+    private static final Set<String> SET_TYPES = Set.of(Set.class.getCanonicalName(), TreeSet.class.getCanonicalName(), HashSet.class.getCanonicalName(), LinkedHashSet.class.getCanonicalName(), Collection.class.getCanonicalName());
```

Comment:
Could you please elaborate why is the `Collection` considered as set type here?

Reply:
Is there a value of distinguishing between `Collection` and `Set`?
If not (and I don't see it here), I would prefer keeping it as is.